compendium/DeclarativeServices/src/ComponentContextImpl.cpp

```diff
@@ -250,30 +250,30 @@ namespace cppmicroservices
         }
 
         std::shared_ptr<void>
-        ComponentContextImpl::LocateService(std::string const& name, std::string const& type) const
+        ComponentContextImpl::LocateService(std::string const& refName, std::string const& type) const
         {
             auto const configManagerPtr = configManager.lock();
             if (!configManagerPtr)
             {
                 throw ComponentException("Context is invalid");
             }
             auto boundServicesCacheHandle = boundServicesCache.lock();
-            auto serviceMapItr = boundServicesCacheHandle->find(name);
+            auto serviceMapItr = boundServicesCacheHandle->find(refName);
             if (serviceMapItr != boundServicesCacheHandle->end())
             {
                 if (auto& services = serviceMapItr->second; !services.empty())
                 {
                     auto& interfaceMapPtr = std::get<1>(services[0]);
 
-                    return GetServicePointer(configManagerPtr, interfaceMapPtr, name, type);
+                    return GetServicePointer(configManagerPtr, interfaceMapPtr, refName, type);
                 }
             }
 
             return nullptr;
         }
 
         std::shared_ptr<void>
-        ComponentContextImpl::LocateService(std::string const& name, cppmicroservices::ServiceReferenceBase const& sRef)
+        ComponentContextImpl::LocateService(std::string const& name, cppmicroservices::ServiceReferenceBase const& sRef) const
```

Comment:
is this `name` parameter a reference name? If so, I recommend changing the name of parameter to `refName`